terminus-ui/expansion-panel/src/expansion-panel.component.ts

```diff
@@ -0,0 +1,322 @@
+  distinctUntilChanged,
+  filter,
+  startWith,
+  take,
+} from 'rxjs/operators';
+import { Subject } from 'rxjs';
+
+import { tsExpansionPanelAnimations } from './expansion-animations';
+import { TsExpansionPanelContentDirective } from './expansion-panel-content.directive';
+import { TS_ACCORDION, TsAccordionBase } from './accordion/accordion-base';
+
+
+/**
+ * The possible states for a {@link TsExpansionPanelComponent}
+ */
+export type TsExpansionPanelState = 'expanded' | 'collapsed';
+
+
+/**
+ * Object that can be used to override the default options for all of the expansion panels in a module.
+ */
+export interface TsExpansionPanelDefaultOptions {
+  /**
+   * Height of the trigger while the panel is expanded
+   */
+  expandedHeight: string;
+
+  /**
+   * Height of the trigger while the panel is collapsed
+   */
+  collapsedHeight: string;
+
+  /**
+   * Whether the toggle indicator should be hidden
+   */
+  hideToggle: boolean;
+}
+
+
+/**
+ * Injection token that can be used to configure the defalt options for the expansion panel component.
+ */
+export const TS_EXPANSION_PANEL_DEFAULT_OPTIONS = new InjectionToken<TsExpansionPanelDefaultOptions>('TS_EXPANSION_PANEL_DEFAULT_OPTIONS');
+
+/**
+ * Unique ID for each panel trigger ID
+ */
+let nextUniqueId = 0;
+
+
+/**
+ * An expansion panel component to show/hide content
+ *
+ * @example
+ * <ts-expansion-panel
+ *               [hideToggle]="true"
+ *               [isExpanded]="true"
+ *               [isDisabled]="true"
+ *               (opened)="panelOpened()"
+ *               (closed)="panelClosed()"
+ *               (expandedChange)="panelStateChanged($event)"
+ *               (destroyed)="componentDestroyed()"
+ *               (afterCollapse)="collapseAnimationDone"
+ *               (afterExpand)="expandAnimationDone()"
+ * >
+ *               <ts-expansion-panel-trigger>
+ *                 Panel trigger
+ *               </ts-expansion-panel-trigger>
+ *
+ *               Panel content
+ * </ts-expansion-panel>
+ *
+ * <example-url>https://getterminus.github.io/ui-demos-master/components/expansion-panel</example-url>
+ */
+@Component({
+  selector: 'ts-expansion-panel',
+  templateUrl: './expansion-panel.component.html',
+  styleUrls: ['./expansion-panel.component.scss'],
+  // NOTE: @Outputs are defined here rather than using decorators since we are extending the @Outputs of the base class
+  // tslint:disable: use-output-property-decorator
+  outputs: ['opened', 'closed', 'expandedChange', 'destroyed'],
+  animations: [tsExpansionPanelAnimations.bodyExpansion],
+  host: {
+    class: 'ts-expansion-panel',
+    '[class.ts-expansion-panel--expanded]': 'expanded',
+    '[class.ts-expansion-panel--animation-noopable]': 'animationMode === "NoopAnimations"',
+  },
+  providers: [
+    // Provide TsAccordionComponent as undefined to prevent nested expansion panels from registering to the same accordion.
+    {
+      provide: TS_ACCORDION,
+      useValue: undefined,
```

Comment:
That's actually a great question.. I honestly didn't try without as Material explicitly uses `undefined`.  My gut says that Angular probably has some internal check since the `useValue` key isn't required. So they likely ignore it if it's not set (meaning completely omitted). But in fairness I have not tested that